cognite/neat/workflows/steps/lib/graph_contextualization.py

```diff
@@ -0,0 +1,144 @@
+    description = "The step matches entities in the graph and creates links based on provided configurations"
+    category = "contextualization"
+    configurables: ClassVar[list[Configurable]] = [
+        Configurable(name="source_class", value="", label="Name of the source class"),
+        Configurable(name="source_property", value="", label="Name of the source property"),
+        Configurable(
+            name="source_value_type",
+            value="single_value_str",
+            label="Type of the value in the source property.Supported options : single_value_str, multi_value_str",
+            options=["single_value_str", "multi_value_str"],
+        ),
+        Configurable(name="target_class", value="", label="Name of the target class"),
+        Configurable(name="target_property", value="", label="Name of the target property"),
+        Configurable(name="relationship_name", value="link", label="Label of the relationship to be created"),
+        Configurable(
+            name="link_direction",
+            value="target_to_source",
+            label="Direction of the relationship. Supported options : target_to_source, source_to_target",
+        ),
+        Configurable(
+            name="matching_method",
+            value="regexp",
+            label="Method to be used for matching. Supported options : exact_match, regexp",
+            options=["exact_match", "regexp"],
+        ),
+        Configurable(
+            name="graph_name",
+            value="source",
+            label="The name of the graph to be used for matching. Supported options : source, solution",
+            options=["source", "solution"],
+        ),
+    ]
+
+    def run(self, graph_store: SolutionGraph | SourceGraph) -> FlowMessage:
+        # We can't use the graph_store to get the graph as input parameter directly,
+        # resolver might resolve the wrong graph
+        # if both are present in the flow context
+        graph_name = self.configs["graph_name"]
+        if graph_name == "solution":
+            graph_store = self.flow_context["SolutionGraph"]
+        else:
+            graph_store = self.flow_context["SourceGraph"]
+        self.ns = PREFIXES["neat"]
+        self.graph_store = graph_store.graph
+        new_links_counter = self.simple_entity_matcher(
+            source_class=self.configs["source_class"],
+            source_property=self.configs["source_property"],
+            source_value_type=self.configs["source_value_type"],
+            target_class=self.configs["target_class"],
+            target_property=self.configs["target_property"],
+            relationship_name=self.configs["relationship_name"],
+            link_direction=self.configs["link_direction"],
+            matching_method=self.configs["matching_method"],
+        )
+        output_text = f"Matcher has created {new_links_counter} links"
+        return FlowMessage(output_text=output_text)
+
+    def simple_entity_matcher(
```

Comment:
I suggest having this methods somewhere in the graph module. I think steps should just be small wrappers around the core functionality.

Reply:
Good catch , at least that is true for NEAT provided steps.